internal/bufferedread/buffered_reader.go

```diff
@@ -369,6 +383,12 @@ func (p *BufferedReader) ReadAt(ctx context.Context, inputBuf []byte, off int64)
 		}
 	}
 
+	resp.Data = data
+	resp.Callback = func() {
+		for _, cb := range callbacks {
+			cb()
```

Comment:
Why extra function calls?

Instead:
1. Change the p.callback() to accept the list of block entry.
2. Instead of creating the slice of callbacks, create slice of entries.
3. Return a final callback which calls p.callbacks(entries).